lib/SparkPost/SparkPost.php

```diff
@@ -107,10 +109,17 @@ public function setHttpAdapter($httpAdapter) {
 
   /**
    * Allows the user to pass in values to override the defaults and set their API key
-   * @param Array $settingsConfig - Hashmap that contains config values for the SDK to connect to SparkPost
+   * @param String | Array $settingsConfig - Hashmap that contains config values
+   *              for the SDK to connect to SparkPost. If its a string we assume that
+   *              its just they API Key.
    * @throws \Exception
    */
-  public function setConfig(Array $settingsConfig) {
+  public function setConfig($settingsConfig) {
+    // if the config map is a string we should assume that its an api key
+    if (gettype($settingsConfig) === 'string') {
```

Comment:
`if (is_string($settingsConfig)) {` ?

The `gettype` docs happen to advise using `get_*` methods for type checking, is the only reason I bring it up:

> Returns the type of the PHP variable var. For type checking, use is_* functions.

http://php.net/manual/en/function.gettype.php